bin/geonetwork_broken_link_checker.py

```diff
@@ -0,0 +1,166 @@
+            continue
+        logger.info("Checking link {0}".format(url))
+        try:
+            request = requests.get(url, data={}, headers={})
+            if request.status_code != 200:
+                broken_urls.append(url)
+                logger.error("Broken link {0}".format(url))
+        except:
+            broken_urls.append(url)
+            logger.error("Broken link {0}".format(url))
+            logger.error("Unexpected error: {0}".format(sys.exc_info()[0]))
+
+    return broken_urls
+
+
+def prepare_output_message():
+    text = list()
+    text.append('')
+    text.append("Metadata tested: {0}".format(len(all_metadata)))
+    text.append("Metadata with broken urls: {0}".format(len(broken_metadata)))
+    text.append("Total tested links: {0}".format(len(all_tested_links)))
+    text.append("Total broken links: {0}".format(len(all_broken_links)))
+
+    for metadata_entry in broken_metadata:
+        text.append('')
+        text.append("Metadata UUID: {0}".format(metadata_entry.uuid))
+        text.append("Metadata URL: {0}".format(metadata_entry.url))
+        text.append("Broken Links: {0}".format(len(metadata_entry.broken_links)))
+        text.append('')
+
+    text.extend(all_broken_links)
+    text.append('')
+
+    return os.linesep.join(text)
+
+
+def broken_link_handler(geonetwork_url):
+    search_url = geonetwork_url + SEARCH_URL_SUFFIX
+    search_result = requests.post(search_url).text
+    root = ET.fromstring(search_result)
+
+    for metadata_xml_entry in root.iter('metadata'):
+        metadata_info = list(metadata_xml_entry)[1]
+        uuid = metadata_info.find('uuid').text
+        metadata_url = "{0}{1}{2}".format(geonetwork_url, METADATA_URL_SUFFIX, uuid)
+        metadata_result = requests.post(metadata_url).text
+        urls = re.findall(URL_REGEX, metadata_result)
+
+        metadata = Metadata(uuid, metadata_url, urls)
+        all_metadata.add(metadata)
+
+    logger.info('Starting broken links check')
+    # count = 0  # Enable to test one metadata
+
+    # count = 0
+    for metadata_entry in all_metadata:
+        # if count == 1:  # Enable to test one metadata
+        #     break  # Enable to test one metadata
+        # else:
+        #     count += 1
+        logger.info("{0} Testing metadata {1} {2}".format('-' * 32, metadata_entry.url, '-' * 32))
+        # All links in metadata
+        links = set(metadata_entry.links)
+
+        # Remove already tested links which will include broken links
+        test_links = links.difference(all_tested_links)
+
+        broken_links = link_checker(test_links)
+        if len(broken_links):
+            metadata_entry.broken_links = broken_links
+            broken_metadata.add(metadata_entry)
+            all_broken_links.update(broken_links)
+
+        all_tested_links.update(test_links)
+
+    logger.info('Finished broken links check')
+    return broken_metadata
+
+
+@click.command()
```

Comment:
I'd replace the line above with
```python
CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
@click.command(context_settings=CONTEXT_SETTINGS)
```

With this in place, someone can type
```bash
python geonetwork_broken_link_checker.py -h
Usage: geonetwork_broken_link_checker.py [OPTIONS]

  Find broken links in geonetwork metadata.

Options:
  --geonetwork-url TEXT  Geonetwork URL. Example: http://catalogue-
                         systest.aodn.org.au/geonetwork  [required]
  --output-file TEXT     File name for output.
  --email-to TEXT        Email the results to.
  --email-from TEXT      Email the results from.
  --mail-server TEXT     Mail Server to send emails. Example:
                         postoffice.sandybay.utas.edu.au:25
  -h, --help             Show this message and exit.
```

Otherwise, someone still need to look at the source code to know what to do